pootle/apps/pootle_store/models.py

```diff
@@ -302,6 +324,26 @@ class Unit(models.Model, base.TranslationUnit):
     unitid_hash = models.CharField(max_length=32, db_index=True,
                                    editable=False)
 
+    # Denormalized from Store.pootle_path
+    # any changes to the `pootle_path` field may require updating the schema
+    # see migration 0007_case_sensitive_schema.py
+    pootle_path = models.CharField(
+        max_length=255, null=False, unique=False,
+        db_index=True, verbose_name=_("Path"))
+
+    # denormalized TP.language
+    language = models.ForeignKey(
+        "pootle_language.Language",
+        null=False,
+        db_index=True,
+        related_name='units')
+    # denormalized TP.project
+    project = models.ForeignKey(
+        "pootle_project.Project",
+        null=False,
```

Comment:
This doesn't allow `NULL` values, but in `016_denormalize_project` migration you are explicitly skipping stores with no project, thus if any of those stores has some unit, the project field for those units will be NULL. Is this not a problem?